src/firstvoices/backend/models/dictionary.py

```diff
@@ -140,13 +142,44 @@ class TypeOfDictionaryEntry(models.TextChoices):
         related_name="incoming_related_dictionary_entries",
     )
 
+    # from fvcharacter:related_words
+    related_characters = models.ManyToManyField(
+        Character,
+        blank=True,
+        through="DictionaryEntryRelatedCharacter",
+        related_name="dictionary_entries",
+    )
+
     class Meta:
         verbose_name = _("Dictionary Entry")
         verbose_name_plural = _("Dictionary Entries")
 
     def __str__(self):
         return self.title
 
+    def save(self, *args, **kwargs):
+        self.clean_confusables()
+        self.set_custom_order()
+        super().save(*args, **kwargs)
+
+    def clean_confusables(self):
+        mapper = AlphabetMapper.objects.filter(site_id=self.site_id).first()
+        confusables_transducer = mapper.preprocess_transducer if mapper else {}
+
+        # strip whitespace and normalize
+        self.title = self.title.strip()
+        self.title = nfc(self.title)
+
+        # clean confusables
+        cleaned_title = confusables_transducer(self.title).output_string
```

Comment:
I would like to update this so that a mapper for the site is created on first access if possible. The mapper needs to exist in order to run the sort process on any entry.